planner/cascades/transformation_rules.go

```diff
@@ -52,18 +52,25 @@ var defaultTransformationMap = map[memo.Operand][]Transformation{
 
 var patternMap = make(map[Transformation]*memo.Pattern)
 
+// GetPattern returns the Pattern of the given Transformation rule.
+// It returns the cached Pattern if possible. Otherwise, generate a new Pattern.
+func GetPattern(r Transformation) *memo.Pattern {
+	if p, ok := patternMap[r]; ok {
```

Comment:
It seems this may cause a concurrent error. We may need to lock this function.
Any better ideas?